cmd/rpcdaemon/commands/get_receipts.go

```diff
@@ -73,6 +77,101 @@ func (api *APIImpl) GetLogs(ctx context.Context, hash common.Hash) ([][]*types.L
 	return logs, nil
 }
 
+// Filter can be used to retrieve and filter logs.
+type Filter struct {
+	addresses []common.Address
+	topics    [][]common.Hash
+
+	block      common.Hash // Block hash if filtering a single block
+	begin, end int64       // Range interval if filtering multiple blocks
+
+	matcher *bloombits.Matcher
+}
+
+func NewBlockFilter(block common.Hash, addresses []common.Address, topics [][]common.Hash) *Filter {
+	// Create a generic filter and convert it into a block filter
+	filter := newFilter(addresses, topics)
+	filter.block = block
+	return filter
+}
+
+// newFilter creates a generic filter that can either filter based on a block hash,
+// or based on range queries. The search criteria needs to be explicitly set.
+func newFilter(addresses []common.Address, topics [][]common.Hash) *Filter {
+	return &Filter{
+		addresses: addresses,
+		topics:    topics,
+	}
+}
+
+// GetLogs returns logs matching the given argument that are stored within the state.
+func (api *APIImpl) GetLogs(ctx context.Context, crit filters.FilterCriteria) ([]*types.Log, error) {
+	var filter *Filter
+	if crit.BlockHash != nil {
+		// Block filter requested, construct a single-shot filter
+		filter = NewBlockFilter(*crit.BlockHash, crit.Addresses, crit.Topics)
+	} else {
+		// Convert the RPC block numbers into internal representations
+		latest, err := getLatestBlockNumber(api.dbReader)
+		if err != nil {
+			return nil, err
+		}
+
+		begin := int64(latest)
+		if crit.FromBlock != nil {
+			begin = crit.FromBlock.Int64()
+		}
+		end := int64(latest)
+		if crit.ToBlock != nil {
+			end = crit.ToBlock.Int64()
+		}
+
+		if begin > end {
```

Comment:
Are you sure we need to error out when being == end or just return empty result?

Reply:
I would check pre-existing behaviour

Reply:
As far as I see go-ethereum just ignores this case.